src/App.tsx

```diff
@@ -0,0 +1,68 @@
+import { useNavigate } from "react-router-dom";
+import "./index.css";
+import "./App.css";
+
+import { useState, useRef } from "react";
+import Modal from "@/components/form/modal";
+import ControlledForm from "@/components/form/ControledForm";
+import UncontrolledForm from "@/components/form/UnControledForm";
+import { useAppSelector } from "@/app/appHook";
+
+import { CardList } from "@/features/user/CardList";
+
+const App: React.FC = () => {
+  const navigate = useNavigate();
+  const [isOpen, setIsOpen] = useState(false);
+  const [formType, setFormType] = useState<"control" | "uncontrol">("control");
+  const result = useAppSelector((state) => state.user);
+  const results = result.data;
+  const lastFocusedButton = useRef<HTMLButtonElement | null>(null);
+
+  const controlFormHandle = (e: React.MouseEvent<HTMLButtonElement>) => {
+    lastFocusedButton.current = e.currentTarget;
+    navigate("/control", { replace: true });
+    setFormType("control");
+    setIsOpen(true);
+  };
+
+  const unControlFormHandle = (e: React.MouseEvent<HTMLButtonElement>) => {
+    lastFocusedButton.current = e.currentTarget;
+    navigate("/uncontrol", { replace: true });
+    setFormType("uncontrol");
+    setIsOpen(true);
+  };
+
+  const doCloseHandle = () => {
+    setIsOpen(false);
+    navigate("/", { replace: true });
+    lastFocusedButton?.current?.focus();
+  };
```

Comment:
The prefix `do` is often redundant. The word `handle` is also generally used to describe event handlers, but in this case, the function is just closing the modal.
You could rename it to `handleCloseModal` or simply `closeModal`.

```suggestion
  const closeModal = () => {
    setIsOpen(false);
    navigate("/", { replace: true });
    lastFocusedButton?.current?.focus();
  };
```